core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -264,41 +287,100 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     const response = await this.threadDataBackendApiService.fetchMessagesAsync(
       threadId);
     const threadMessageBackendDicts = response.messages;
-    this.reviewMessage = threadMessageBackendDicts.map(
-      m => ThreadMessage.createFromBackendDict(m))[1].text;
+    let threadMessages = threadMessageBackendDicts.map(
+      m => ThreadMessage.createFromBackendDict(m));
+    // This is to prevent a console error when a contribution
+    // doesn't have a review message. When a contribution has
+    // a review message the second element of the threadMessages
+    // array contains the actual review message.
+    if (threadMessages[1] !== undefined) {
+      this.reviewMessage = threadMessages[1].text;
+    }
   }
 
-  showNextItemToReview(suggestionId: string): void {
-    this.resolvedSuggestionIds.push(this.activeSuggestionId);
-    if (this.lastSuggestionToReview) {
+  refreshModalData(): void {
+    this.activeSuggestion = this.activeContribution.suggestion;
+    this.activeContributionDetails = this.activeContribution.details;
+    this.contextService.setCustomEntityContext(
+      AppConstants.IMAGE_CONTEXT.EXPLORATION_SUGGESTIONS,
+      this.activeSuggestion.target_id);
+    this.subheading = (
+      `${this.activeContributionDetails.topic_name} / ` +
+      `${this.activeContributionDetails.story_title} / ` +
+      `${this.activeContributionDetails.chapter_title}`
+    );
+    this.init();
```

Comment:
This is still confusing, why you call `init` inside refresh. Can you do it the other way around?